File: CS/WindowsApplication1/MyTreeList.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.Collections;
using DevExpress.XtraTreeList;

namespace WindowsApplication1
{
    public class MyTreeList : TreeList
    {

        public MyTreeList()
        {

        }

        protected override string GetColumnError(DevExpress.XtraTreeList.Columns.TreeListColumn column, DevExpress.XtraTreeList.Nodes.TreeListNode node)
        {
          if (column == null || column.FieldName == "Unbound") return string.Empty;// your condition 
            return base.GetColumnError(column, node);
        }

    }
}

File: VB/WindowsApplication1/MyTreeList.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports System.Collections
Imports DevExpress.XtraTreeList

Namespace WindowsApplication1
	Public Class MyTreeList
		Inherits TreeList

		Public Sub New()

		End Sub

		Protected Overrides Overloads Function GetColumnError(ByVal column As DevExpress.XtraTreeList.Columns.TreeListColumn, ByVal node As DevExpress.XtraTreeList.Nodes.TreeListNode) As String
		  If column Is Nothing OrElse column.FieldName = "Unbound" Then ' your condition
			  Return String.Empty
		  End If
			Return MyBase.GetColumnError(column, node)
		End Function

	End Class
End Namespace
